File: python/p9.py
from sys import argv;
class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
       	if x < 0:
		return False;
	if x == 0:
		return True;
	v = [];
	while x > 0:
		v.append(x%10);
		x = x/10;
	l = len(v);
	for i in range(l):
		if v[i] != v[l-i-1]:
			return False;
	return True;

if __name__ == "__main__":
	x = int(argv[1]);
	sol = Solution();
	print sol.isPalindrome(x);

File: c++/p9.cpp
#include<iostream>
#include<vector>
using namespace std;
class Solution {
public:
    bool isPalindrome(int x) {
	if (x < 0){
		return false;
	}
	if (x == 0){
		return true;
	};
	vector<int> l;
	while (x != 0){
		l.push_back(x%10);
		x /= 10;
	};
	int len = l.size();
	for (int i = 0; i < len; ++i){
		if (l[i] != l[len-i-1]){
			return false;
		}
	}
	return true;
    }
};
int main(int argc, char** argv){
	Solution sol;
	cout<<sol.isPalindrome(atoi(argv[1]))<<endl;
	return 0;
}
